### PR TITLE
feat(abstractda): add val manipulation capability

### LINKED ISSUE #274: Add `Val` manipulation capanbility to BDA and DA wizard

### LINKED ISSUE #279: feat(da): add dchg, qchg and dupd to da wizard

### LINKED ISSUE #280: refactor(DAType and DOType wizards): move da and bda wizards to library

### PR DESCRIPTION
**Pull request by JakobVogelsang**
Closes #274 
Closes #280 
Closes #279 

This PR does not only add basic `Val` manipulation capability but also `dchg`, `qchg` and `dupd` - last three attributes of `DA`. I also found the pattern for the name of both `BDA` and `DA` was not correct.
I also take the chance and pushed the touched wizards to the wizard library.

### CODE REVIEW

File: src/wizard-select.ts [hunk 1]
```diff
@@ -72,14 +72,7 @@ export class WizardSelect extends Select {
   }
 
   checkValidity(): boolean {
-    if (
-      this.reservedValues &&
-      this.reservedValues.some(array => array === this.value)
-    ) {
-      this.setCustomValidity(get('textfield.unique'));
```
Comment (ca-d) [inline src/wizard-select.ts:79]: Great change 🎉 ! Does this mean we could also get rid of the `textfield.unique` translation, or do we still need it in the `wizard-textfield`?

File: src/wizards/abstractda.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    html`<wizard-textfield
+      label="Val"
+      .maybeValue=${Val}
+      helper="${translate('scl.Val')}"
+      nullable
+    ></wizard-textfield>`,
+  ];
+}
+
+export function getValAction(
```
Comment (ca-d) [inline src/wizards/abstractda.ts:211]: Do you generally prefix functions like this with `get` or would

```ts
export function valAction(
```

also work for you?

File: src/wizards/bda.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+      else newElement.setAttribute('desc', desc);
+      newElement.setAttribute('bType', bType);
+      if (type === null) newElement.removeAttribute('type');
+      else newElement.setAttribute('type', type);
+      if (sAddr === null) newElement.removeAttribute('sAddr');
+      else newElement.setAttribute('sAddr', sAddr);
+      if (valKind === null) newElement.removeAttribute('valKind');
+      else newElement.setAttribute('valKind', valKind);
+      if (valImport === null) newElement.removeAttribute('valImport');
+      else newElement.setAttribute('valImport', valImport);
```
Comment (ca-d) [inline src/wizards/bda.ts:63]: Maybe we could think about refactoring blocks like this into a helper function in our `foundation` in the future. How common do you think these if/else blocks are going to be?

File: src/wizards/foundation.ts/enums.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+export const functionalConstraintEnum = [
```
Comment (ca-d) [inline src/wizards/foundation.ts/enums.ts:1]: Why does the `wizards/foundation` directory carry a `.ts` extension?

**Review by ca-d — CHANGES_REQUESTED** [review]
This looks and reads extremely well. The only thing that sticks out to me is the `foundation.ts` directory name which looks like a TypeScript file.

File: src/wizard-select.ts (hunk 1 above)
Comment (JakobVogelsang) [reply src/wizard-select.ts:79]: At the moment, we still use it to check for name attribute duplication.

File: src/wizards/abstractda.ts (hunk 2 above)
Comment (JakobVogelsang) [reply src/wizards/abstractda.ts:211]: We do this at the moment for `Voltage` for example and will do so for `MinTime`, `MaxTime` ....
Those are child elements  that in our wizards appear as attributes. 

Working on the wizards, I found that we should talk about a naming convention here. I would therefore postpone this to [Issue](https://github.com/openscd/open-scd/issues/281)

File: src/wizards/bda.ts (hunk 3 above)
Comment (JakobVogelsang) [reply src/wizards/bda.ts:63]: This is very true. This pattern is very common and can be avoided with a simple function [Issue](https://github.com/openscd/open-scd/issues/282)

File: src/wizards/foundation.ts/enums.ts (hunk 4 above)
Comment (JakobVogelsang) [reply src/wizards/foundation.ts/enums.ts:1]: This is a typo, sorry :)

**Review by ca-d — APPROVED** [review]
LGTM

**Pull request merged**